src/components/SingleArticlePage.jsx

```diff
@@ -0,0 +1,35 @@
+import { useParams } from "react-router"
+import { useState, useEffect } from "react"
+import { Link } from "react-router-dom"
+import { fetchArticleById } from "./api"
+
+function SingleArticlePage() {
+    const {article_id} = useParams()
+    const [article, setArticle] = useState([])
+
+    useEffect(() => {
+        fetchArticleById(article_id).then((article) => {
+            setArticle(article.data.article)
+        })
+        .catch((err) => console.log(err))
+    }, [article_id])
+
+    if(!article){
+        return <p>Loading...</p>
```

Comment:
What if there is an article but the user wants to load up a different one/ change the article id?